src/pitaya_core/src/metrics.rs

```diff
@@ -133,16 +133,31 @@ pub fn exponential_buckets(start: f64, factor: f64, count: usize) -> Vec<f64> {
     buckets
 }
 
-pub async fn record_histogram_duration(
+pub async fn record_histogram_duration<'a>(
+    logger: slog::Logger,
     reporter: ThreadSafeReporter,
-    name: &str,
+    name: &'a str,
     start: std::time::Instant,
-    labels: &[&str],
+    labels: &'a [&'a str],
 ) {
     let elapsed = std::time::Instant::now() - start;
-    reporter
+    if let Err(e) = reporter
         .read()
         .await
         .observe_hist(name, elapsed.as_secs_f64(), labels)
-        .expect("should not fail to observe");
+    {
+        slog::warn!(logger, "observe_hist failed"; "err" => %e);
+    }
+}
+
+pub async fn add_gauge<'a>(
```

Comment:
Shouldn't you use `register_gauge` instead? We were deprecating the `AddGauge` in NPitaya and using the `Register` prefix.

Reply:
This should be `add_to_gauge` in order to make it explicit that we're not registering the metric here, but observing it.